analyzer/repo-cloner-plugin/src/main/java/eu/fasten/analyzer/repoclonerplugin/RepoClonerPlugin.java

```diff
@@ -138,14 +140,16 @@ public void consume(String record) {
                     logger.info("Successfully cloned the repository for " + group + ":" + artifact + ":" + version + " from " + repoUrl);
                 } else if (result.getSecond() != null) {
                     var errorTriple = result.getSecond();
-                    logger.error("Could not clone the repository for " + group + ":" + artifact + ":" + version + " from " + repoUrl
+                    var errMsg = "Could not clone the repository for " + group + ":" + artifact + ":" + version + " from " + repoUrl
                             + "; GitCloner: " + (errorTriple.getLeft() != null ? errorTriple.getLeft().getMessage() : "null")
                             + "; SvnCloner: " + (errorTriple.getMiddle() != null ? errorTriple.getMiddle().getMessage() : "null")
-                            + "; HgCloner: " + (errorTriple.getRight() != null ? errorTriple.getRight().getMessage() : "null")
-                    );
+                            + "; HgCloner: " + (errorTriple.getRight() != null ? errorTriple.getRight().getMessage() : "null");
+                    logger.error(errMsg);
+                    this.pluginError = new CloneFailedException(errMsg);
                 }
             } else {
                 logger.info("Repository URL not found");
+                this.pluginError = new RepoUrlNotFoundException("Repository URL not found");
```

Comment:
Could you please include the URL in the message(s). This would allow us to identify patterns in the failing URLs...

Reply:
Actually, there is NO repo URL in the else block. But the failing records are included in `.err` records.